packages/acceptance-tests/pkg-tests-specs/sources/commands/set/version.test.ts

```diff
@@ -136,20 +136,76 @@ describe(`Commands`, () => {
+        await xfs.writeFilePromise(ppath.join(path, Filename.rc), `yarnPath: ${yarnPath}`);
+        await run(`set`, `version`, `3.0.0`);
+        await check(path, {corepackVersion: `3.0.0`, usePath: true}, ppath.dirname(yarnAbsPath));
+      }),
+    );
+
+    test(
+      `it should respect the current yarnPath, if it is defined`,
+      makeTemporaryEnv({}, {
+        env: {COREPACK_ROOT: undefined},
+      }, async ({path, run, source}) => {
+        await run(`set`, `version`, `4.0.0`);
+        await check(path, {corepackVersion: `4.0.0`, usePath: true});
+        const oldYarnPath = ppath.join(path, `.yarn/releases/yarn-4.0.0.cjs`);
+        const releasesPPath = ppath.join(path, `releases_dir`);
+        await xfs.mkdirPromise(releasesPPath);
+        const yarnAbsPath = ppath.join(releasesPPath, `/yarn-4.0.0.cjs`);
+        const yarnPath = ppath.relative(path, yarnAbsPath);
+        await xfs.renamePromise(oldYarnPath, yarnAbsPath);
+        await xfs.writeFilePromise(ppath.join(path, Filename.rc), `yarnPath: ${yarnPath}`);
+        await run(`set`, `version`, `3.0.0`);
+        await check(path, {corepackVersion: `3.0.0`, usePath: true}, ppath.dirname(yarnAbsPath));
+      }),
+    );
+    test(
+      `it should respect the current yarnPath if defined, even when corepack is enabled and the version range is semver`,
+      makeTemporaryEnv({}, {
+        env: {COREPACK_ROOT: undefined},
+      }, async ({path, run, source}) => {
+        await run(`set`, `version`, `4.0.0`);
+        await check(path, {corepackVersion: `4.0.0`, usePath: true});
+        const oldYarnPath = ppath.join(path, `.yarn/releases/yarn-4.0.0.cjs`);
+        const releasesPPath = ppath.join(path, `releases_dir`);
+        await xfs.mkdirPromise(releasesPPath);
+        const yarnAbsPath = ppath.join(releasesPPath, `/yarn-4.0.0.cjs`);
+        const yarnPath = ppath.relative(path, yarnAbsPath);
+        await xfs.renamePromise(oldYarnPath, yarnAbsPath);
+        await xfs.writeFilePromise(ppath.join(path, Filename.rc), `yarnPath: ${yarnPath}`);
+        await run(`set`, `version`, `3.0.0`, {env: {COREPACK_ROOT: `/path/to/corepack`}});
+        await check(path, {corepackVersion: `3.0.0`, usePath: true}, ppath.dirname(yarnAbsPath));
       }),
     );
   });
 });
 
-async function check(path: PortablePath, checks: {corepackVersion: string | RegExp, usePath: boolean}) {
-  const releasesPath = ppath.join(path, `.yarn/releases`);
+async function check(path: PortablePath, checks: {corepackVersion: string | RegExp, usePath: boolean}, releasesDir?: PortablePath) {
+  releasesDir = checks.usePath ? (releasesDir ?? ppath.join(path, `.yarn/releases`)) : undefined;
   const yarnrcPath = ppath.join(path, Filename.rc);
   const manifestPath = ppath.join(path, Filename.manifest);
 
   let releases: Array<string> | null;
   try {
-    releases = await xfs.readdirPromise(releasesPath);
+    releases = releasesDir ? await xfs.readdirPromise(releasesDir) : null;
```

Comment:
Since the releases dir path is no longer hardcoded, I figured I should change the check function, in order to make the releases dir path a parameter, rather than a constant.

Reply:
Also, my first ideia was to retrieve the releases dir path from the .`yarnrc.yml` file, but that seemed to make the tests super slow. So I made it a parameter.